test/test_metric/test_metrics.py

```diff
@@ -326,6 +75,185 @@ def test_sign_flip(self):
         self.assertAlmostEqual(score, -1.0)
 
 
+# Trial changes for _PredictScorer, _ProbaScorer, and _TresholdScorer.
+@pytest.mark.parametrize(
+    "y_pred, y_true, scorer, expected_score",
+    [
+        (
+            np.array([[1.0, 0.0], [1.0, 0.0], [0.0, 1.0], [0.0, 1.0]]),
+            np.array([0, 0, 1, 1]),
+            autosklearn.metrics.accuracy,
+            1.0,
+        ),
+        (
+            np.array([[0.0, 0.0], [0.0, 0.0], [0.0, 0.0], [0.0, 0.0]]),
+            np.array([0, 0, 1, 1]),
+            autosklearn.metrics.accuracy,
+            0.5,
+        ),
+        (
+            np.array([[1.0, 1.0], [1.0, 1.0], [1.0, 1.0], [1.0, 1.0]]),
+            np.array([0, 0, 1, 1]),
+            autosklearn.metrics.balanced_accuracy,
+            0.5,
+        ),
+        (
+            np.array([[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 1.0]]),
+            np.array([0, 1, 2]),
+            autosklearn.metrics.accuracy,
+            1.0,
+        ),
+        (
+            np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0]]),
+            np.array([0, 1, 2]),
+            autosklearn.metrics.accuracy,
+            0.333333333,
+        ),
+        (
+            np.array([[1.0, 1.0, 1.0], [1.0, 1.0, 1.0], [1.0, 1.0, 1.0]]),
+            np.array([0, 1, 2]),
+            autosklearn.metrics.accuracy,
+            0.333333333,
+        ),
+        (
+            np.array([[1.0, 1.0, 1.0], [1.0, 1.0, 1.0], [1.0, 1.0, 1.0]]),
+            np.array([0, 1, 2]),
+            autosklearn.metrics.balanced_accuracy,
+            0.333333333,
+        ),
+        (
+            np.array([[0.0, 0.0], [0.0, 1.0], [1.0, 0.0], [1.0, 1.0]]),
+            np.array([[0, 0], [0, 1], [1, 0], [1, 1]]),
+            autosklearn.metrics.accuracy,
+            1.0,
+        ),
+        (
+            np.array([[0.0, 0.0], [0.0, 0.0], [0.0, 0.0], [0.0, 0.0]]),
+            np.array([[0, 0], [0, 1], [1, 0], [1, 1]]),
+            autosklearn.metrics.accuracy,
+            0.25,
+        ),
+        (
+            np.array([[1.0, 1.0], [1.0, 1.0], [1.0, 1.0], [1.0, 1.0]]),
+            np.array([[0, 0], [0, 1], [1, 0], [1, 1]]),
+            autosklearn.metrics.accuracy,
+            0.25,
+        ),
+        (
+            np.arange(0, 1.01, 0.1),
+            np.arange(0, 1.01, 0.1),
+            autosklearn.metrics.r2,
+            1.0,
+        ),
+        (
+            np.ones(np.arange(0, 1.01, 0.1).shape) * np.mean(np.arange(0, 1.01, 0.1)),
+            np.arange(0, 1.01, 0.1),
+            autosklearn.metrics.r2,
+            0.0,
+        ),
+        (
+            np.array([[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 1.0]]),
+            np.array([0, 0, 1, 1]),
+            autosklearn.metrics.log_loss,
+            0.0,
+        ),
+        (
+            np.array([[0.0, 0.0], [0.0, 0.0], [0.0, 0.0], [0.0, 0.0]]),
```

Comment:
The workflow test fails due to a sign flip in the case of ``log_loss`` as the ``greater_is_better`` hyperparameter in ``autosklearn.metrics.log_loss`` is set as False [as seen here](https://github.com/automl/auto-sklearn/blob/013d7eee3c46f0f0c0f66ab7eac9dd1945faf101/autosklearn/metrics/__init__.py#L388-L395). Hence, to rectify this, should I change ``greater_is_better`` hyperparameter for ``autosklearn.metrics.log_loss`` to True or should I flip the signs in cases where the test fails?

Reply:
I'm not sure which of the cases fail in this particularly test but you can just delete the test cases which rely on the `log_loss` having `greater_is_better=True`, as `log_loss` is something we want to have stay as a minimization metric.